reactnative/RNPlayground/app/navigators/PlaygroundNavigator.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from "react"
+import { BaseScreenProps, NavigatorParamList } from "."
+import { createNativeStackNavigator } from '@react-navigation/native-stack'
+import { StackScreenProps } from "@react-navigation/stack"
+import { PhotosScreen } from "../screens/PhotosScreen"
+import { CompositeScreenProps } from "@react-navigation/native"
+import { PlaygroundScreen } from "../screens/PlaygroundScreen"
+import { Appbar } from "react-native-paper"
+import { colors } from "../theme/colors"
+import { View } from "react-native"
+import { Text } from "../components"
+import { getHeaderTitle } from '@react-navigation/elements'
+import { translate } from "../i18n"
+
+type PlaygroundTabParamList = {
+  Playground: undefined,
+  Photos: undefined
+}
+
+/**
+ * Helper for automatically generating navigation prop types for each route.
+ *
+ * More info: https://reactnavigation.org/docs/typescript/#organizing-types
+ */
+export type PlaygroundTabScreenProps<T extends keyof PlaygroundTabParamList> = CompositeScreenProps<
+  StackScreenProps<PlaygroundTabParamList, T>,
+  BaseScreenProps<keyof NavigatorParamList>
+>
+
+const Stack = createNativeStackNavigator<PlaygroundTabParamList>()
+
+export function PlaygroundNavigator() {
```

Comment:
this is a nested stack navigator inside Playground screen that has an header defined here and each screen can modify it through options. An alternative would be for each screen to have its own header.

I wanted to try this approach because I was curious if defining it here once would change the behavior with regards to how the screens on the stack are being rendered. I was hoping that if the header is defined here then it would stay fixed and the screen transition would contain only Screen content, updating the header state at the end of the transition. But that doesn't seem to be the case, the header seems to be added to each screen so effectively there's no difference between having it here and having it in each screen.